### PR TITLE
Adds warning when client BYOND version is old

### PR DESCRIPTION
**Pull request by Kyep**
Adds a chat text warning when a client connects to the server with a version of BYOND which is at least one major version behind what is current.
E.g we've declared the standard to be 510, and they connect with 509.

This warning does not actually stop them doing anything. It just tells them to go update their BYOND.

This does not support minor versions (e.g: 510.1347) because the client's "byond_version" var only reports the major version number.
Still, should help remind people who return to the game after an absence to update their client before playing.

### CODE REVIEW

**Comment by Tastyfish** [comment]
It's a shame there's no way to check the minor version.

**Comment by monster860** [comment]
Sure you can. Make a lobby-screen sized image that reminds people about updating byond to the latest version, and then put that on the lobby screen at plane 1. Then get an image overlay that has the actual lobby screen on it, and add it as an overlay. That way, it will show in front if PLANE_MASTER is something the client understands.

File: code/modules/client/client procs.dm [hunk 1]
```diff
@@ -258,8 +260,11 @@
 
 	if(connection != "seeker")					//Invalid connection type.
 		return null
-	if(byond_version < MIN_CLIENT_VERSION)		//Out of date client.
+	if(byond_version < MIN_CLIENT_VERSION) // Too out of date to play at all, force update.
+		to_chat(src,"<span class='userdanger'>Your BYOND client (v: [byond_version]) is too far out of date to to play. Get the latest client from http://www.byond.com/")
```
Comment (Krausus) [inline code/modules/client/client procs.dm]: This message will never actually be visible; booting a client in /client/New disconnects them before the BYOND UI even shows up. Using `alert`, as the guest removal does below, doesn't work either. Neither does `src << "message"`...

I'm not sure there's actually any way to send a client a message before disconnecting them here. You might actually have to let them connect, show them the message, and *then* disconnect them shortly after, if that can even be done reliably with how long it takes to load in sometimes.

**Comment by Kyep** [comment]
@Tastyfish
- Yes, it is.

@monster860
- Even assuming that method works, that'll only tell people to update if their version is less than the version that starts properly supporting PLANE_MASTER. It would not help us in a few months when we need to persuade people to update from 5.10 to 5.11. It is only useful once.

@krausus
- There are two version checks. SUGGESTED_CLIENT_VERSION is the new one - the one I am adding. MIN_CLIENT_VERSION is a setting someone else added ages ago, and never used, because it forcibly disconnects anyone with too low a version, and nobody wants to do that.
- My changes to MIN_CLIENT_VERSION add a message to the user where previously there was none at all. Even then, MIN_CLIENT_VERSION is still not enabled, as the goal of this PR is to remind people, not block them. I'm just making a little edit to it so it is a little more usable in the event we ever decide to use it.
- My addition of SUGGESTED_CLIENT_VERSION uses to_chat to notify the user, which I have tested, and I have found *does* work in this context. SUGGESTED_CLIENT_VERSION is the main thing I'm adding here.
- I am not using alert() at all, because I tested it several times, and found it never worked for me. to_chat, on the other hand, did work.

**Comment by Krausus** [comment]
Yes, I know all that. I commented specifically on the message you added for `MIN_CLIENT_VERSION`, which is **never visible**, even if we start using it. My point was, you shouldn't add a message there, because it doesn't work.

**Comment by Kyep** [comment]
@Krausus Changed.

**Pull request merged**